### DISCUSSION TITLE
How to implement a file picker

### DISCUSSION THREAD

**Discussion by manikantagarikipati** [varabyte/kobweb#494] [Q&A]
I have a feature request where i want to implement file picker ( basically pick excel sheet but i assume the same solution works for all file types ) 

however i could find any documentation or any other libraries that helps with this, does anyone knows of how to make it work?

**Comment by bitspittle** [comment]
Check out https://github.com/varabyte/kobweb/blob/c7799cc1d8f5c5125af036531fb17cf429c6a334/frontend/browser-ext/src/jsMain/kotlin/com/varabyte/kobweb/browser/file/FileUtils.kt#L186

As far as I'm aware, note that you don't really have more control than that. The browser, for security reasons, controls the whole file request pipeline. You can just ask it to start the process and then you get a callback when it is done.
